app/services/catalog/order_item_transition.rb

```diff
@@ -0,0 +1,50 @@
+module Catalog
+  class OrderItemTransition
+    attr_reader :order_item_id
+    attr_reader :order_item
+    attr_reader :state
+
+    def initialize(order_item_id)
+      @order_item = OrderItem.find(order_item_id)
+      @approvals = @order_item.approval_requests
+    end
+
+    def process
+      if approved?
+        submit_order
+        @state = "approved"
+      elsif denied?
```

Comment:
@lindgrenj6 just a thing to look out for, the denied might come at different times. If the first workflow says deny and then another one takes a day and marks it as denied we would be marking the order the second time as denied. Not sure if that is an issue. but we should track it in a separate issue.